storage/src/adb/current.rs

```diff
@@ -102,70 +102,82 @@ impl<
         };
 
         let context = context.with_label("adb::current");
-        let mut status = Bitmap::<H, N>::restore_pruned(
+        let mut status = Bitmap::restore_pruned(
             context.with_label("bitmap"),
             &config.bitmap_metadata_partition,
         )
         .await?;
 
         // Initialize the db's mmr/log.
-        let (mmr, log) =
-            Any::<_, _, _, _, T>::init_mmr_and_log(context.clone(), hasher, cfg).await?;
+        let mut hasher = H::new();
+        let mut b_hasher = Basic::new(&mut hasher);
+        let (mut mmr, log) =
+            Any::<_, _, _, _, T>::init_mmr_and_log(context.clone(), &mut b_hasher, cfg).await?;
 
         // Ensure consistency between the bitmap and the db's MMR.
-        let start_leaf_num = leaf_pos_to_num(mmr.pruned_to_pos()).unwrap();
+        let mmr_pruned_pos = mmr.pruned_to_pos();
+        let mut start_leaf_num = leaf_pos_to_num(mmr_pruned_pos).unwrap();
+        let bit_count = status.bit_count();
+        if start_leaf_num < bit_count {
+            // This can happen if the commit operation failed before the mmr was pruned.
+            warn!(
+                start_leaf_num,
+                bit_count, "mmr starting leaf precedes bitmap pruning point"
+            );
+            start_leaf_num = bit_count;
+        }
+
         let pruned_bits = status.pruned_bits();
         let bitmap_pruned_pos = leaf_num_to_pos(pruned_bits);
-        let mmr_pruned_pos = mmr.pruned_to_pos();
         let mmr_pruned_leaves = leaf_pos_to_num(mmr_pruned_pos).unwrap();
-        assert!(
-            bitmap_pruned_pos <= mmr_pruned_pos,
-            "bitmap is pruned beyond where bits should be retained"
-        );
-        if bitmap_pruned_pos < mmr.pruned_to_pos() {
+        let mut g_hasher = GHasher::new(&mut hasher, 9, &mmr);
```

Comment:
Is this meant to be hardcoded as `9`?

Reply:
hah, yeah, was supposed to be log2(CHUNK_SIZE_BITS).  fixed.